transports/p2p/peerstate.go

```diff
@@ -0,0 +1,45 @@
+package p2p
+
+import "time"
+
+// peerState maintains state of inbound, persistent, outbound peers as well
+// as banned peers and outbound groups.
+type peerState struct {
+	inboundPeers    map[int32]*serverPeer
+	outboundPeers   map[int32]*serverPeer
+	persistentPeers map[int32]*serverPeer
+	banned          map[string]time.Time
+	outboundGroups  map[string]int
+	connectionCount map[string]int
+}
+
+// Count returns the count of all known peers.
+func (ps *peerState) Count() int {
+	return len(ps.inboundPeers) + len(ps.outboundPeers) +
+		len(ps.persistentPeers)
+}
+
+// CountIP returns the count of all peers matching the IP.
+func (ps *peerState) CountIP(host string) int {
+	return ps.connectionCount[host]
+}
+
+// forAllOutboundPeers is a helper function that runs closure on all outbound
+// peers known to peerState.
+func (ps *peerState) forAllOutboundPeers(closure func(sp *serverPeer)) {
+	for _, e := range ps.outboundPeers {
+		closure(e)
```

Comment:
The `e` looks to me like an error (like in `catch(e) {  }`).
Consider another name like `peer` for example.

Reply:
Done